### PR TITLE
Declare the cococapods-interactor CLI interface and expose it in the CocoapodsInteractor.swift

### PR DESCRIPTION
**Pull request by pepicrft**
### Short description 📝
Continuing with the work of adding support for Pods to the `Dependencies.swift`, I exposed a CLI interface from the `cocoapods-interactor` Ruby CLI with two commands, `install` and `update`. They'll take a directory where the Podfile and the dependencies are, run `install` or `update`, and export the graph in a format that's parseable by Tuist.

In the next PR I'll work on building the `tuist dependencies install` command for Pods.

**Note:** That the `vendor/thor` directory is a third-party Ruby Gem that I'm vendoring for convenience to declaring the CLI interface. I played with Ruby's component from the standard library but it's very limited.

### Checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase.
- [ ] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [ ] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [ ] In case the PR introduces changes that affect users, the documentation has been updated.
- [ ] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

**Comment by fortmarek** [comment]
> That the vendor/thor directory is a third-party Ruby Gem that I'm vendoring for convenience to declaring the CLI interface. I played with Ruby's component from the standard library but it's very limited.

Out of curiosity I checked the size this will add:
```
-rw-r--r--   1 marekfort  staff    40B  2 Dec 20:40 .sha
drwxr-xr-x  21 marekfort  staff   672B  2 Dec 20:40 thor
-rw-r--r--   1 marekfort  staff    16K  2 Dec 20:40 thor.rb
```

Seems to be limited, indeed 👍

File: Sources/TuistDependencies/Cocoapods/CocoaPodsInteractor.swift [hunk 1]
```diff
@@ -3,17 +3,27 @@ import TSCBasic
...
     private let binaryLocator: BinaryLocating
 
     init(binaryLocator: BinaryLocating = BinaryLocator()) {
         self.binaryLocator = binaryLocator
     }
 
-    func run() throws {
-        let path = try binaryLocator.cocoapodsInteractorPath()
-        try System.shared.runAndPrint(path.pathString)
+    func run() throws {}
```
Comment (fortmarek) [inline Sources/TuistDependencies/Cocoapods/CocoaPodsInteractor.swift:18]: should this be deleted?

**Review by fortmarek — APPROVED** [review]

File: Sources/TuistDependencies/Cocoapods/CocoaPodsInteractor.swift (hunk 1 above)
Comment (pepicrft) [reply Sources/TuistDependencies/Cocoapods/CocoaPodsInteractor.swift]: Yes!

**Review by laxmorek — APPROVED** [review]
> > Note: That the vendor/thor directory is a third-party Ruby Gem that I'm vendoring for convenience to declaring the CLI interface. I played with Ruby's component from the standard library but it's very limited.

👍

**Pull request merged**